src/main/java/racingcar/RacingGame.java

```diff
@@ -20,27 +25,43 @@ public RacingGame(InputView inputView, ResultView resultView, RandomNumberGenera
 
     public void play() {
         inputView.initGameInput();
-        initializeCars(inputView.getNumberOfCars());
+        initializeCars(inputView.getCarNames());
         runRace(inputView.getAttemptCount());
-        resultView.printResult();
+        resultView.printWinners(getWinners(cars));
     }
 
     public List<Car> runRace(int attemptCount) {
+        resultView.printResultHeader();
+        resultView.printResults(cars);
+
         for (int i = 0; i < attemptCount; i++) {
             moveCars();
-            resultView.saveResults(cars);
+            resultView.printResults(cars);
         }
         return cars;
     }
 
-    public void initializeCars(int numberOfCars) {
-        cars = new ArrayList<>();
-        for (int i = 0; i < numberOfCars; i++) {
-            cars.add(new Car());
+    public void initializeCars(String[] carNames) {
+        cars = Arrays.stream(carNames).peek(this::validateCarNameLength).map(Car::new).collect(Collectors.toList());
+    }
+
+    public void validateCarNameLength(String carName) {
+        if (carName.length() > CAR_NAME_LENGTH_LIMIT) {
+            throw new IllegalArgumentException("Car name is too long");
         }
     }
 
     public void moveCars() {
-        cars.forEach(car -> car.move(randomNumberGenerator.generate(10)));
+        cars.forEach(car -> car.move(randomNumberGenerator.generate(RANDOM_NUMBER_BOUNDARY)));
+    }
+
+    public List<Car> getWinners(List<Car> cars) {
+        int maxPosition = cars.stream().mapToInt(Car::getPosition).max().orElse(1);
+
+        return cars.stream().filter(car -> car.getPosition() == maxPosition).collect(Collectors.toList());
```

Comment:
```suggestion
        return cars.stream().filter(car -> car.isEqualPosition(maxPosition)).collect(Collectors.toList());
```

게터를 사용하는 대신 객체에게 메시지를 전달하는 방법은 어떤가요?
메소드 네이밍은 예시이니 참고만 해주세요~